src/Sarif/Core/Tool.cs

```diff
@@ -49,5 +50,16 @@ public static Tool CreateFromAssemblyData(string prereleaseInfo = null)
 
             return tool;
         }
+
+        public bool ShouldSerializeRulesMetadata()
+        {
+            return this.RulesMetadata != null &&
+                    this.RulesMetadata.Where((n) => { return n != null; }).Any();
+        }
+        public bool ShouldSerializeNotificationsMetadata()
+        {
+            return this.NotificationsMetadata != null &&
+                    this.NotificationsMetadata.Where((n) => { return n != null; }).Any();
+        }
```

Comment:
>} [](start = 8, length = 1)

You can DRY these two methods out:
```
public bool ShouldSerializeRulesMetadata() => ShouldSerializeMessageDescriptors(this.RulesMetadata);
public bool ShouldSerializeNotificationsMetadata() => ShouldSerializeMessageDescriptors(this.NotificationsMetadata);

static private bool ShouldSerializeMessageDescriptors(IEnumerable<MessageDescriptor> messageDescriptors)
    =>messageDescriptors?.Any(n => n != null) == true;
```

Reply:
For two one-liners only? Not super compelling. But! a generic helper for IEnumerable<T> s/be usable in lots of places. I see something that already exists for IList<string> that could be generalized. BUT! This code takes advantage of the trick that an extension method can be passed (and properly handle) a null 'this' instance. This avoids the need for the explicit null check in the caller but breaks the principle of least surprise. still might be worth it. take a look when i push.

---
In reply to: [255307231](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255307231) [](ancestors = 255307231)

Reply:
Yes, even for two-liners. Consider:

1. I had a comment on your implementation. You had to change it in two places.
2. The reader has to convince himself that the two implementations _are_ the same. My formulation makes that explicit; no thought required.

---
In reply to: [255311391](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255311391) [](ancestors = 255311391,255307231)